cmd/syncthing/gui.go

```diff
@@ -270,10 +269,12 @@ func (s *apiService) Serve() {
 	postRestMux.HandleFunc("/rest/system/restart", s.postSystemRestart)        // -
 	postRestMux.HandleFunc("/rest/system/shutdown", s.postSystemShutdown)      // -
 	postRestMux.HandleFunc("/rest/system/upgrade", s.postSystemUpgrade)        // -
-	postRestMux.HandleFunc("/rest/system/pause", s.postSystemPause)            // device
-	postRestMux.HandleFunc("/rest/system/resume", s.postSystemResume)          // device
 	postRestMux.HandleFunc("/rest/system/debug", s.postSystemDebug)            // [enable] [disable]
 
+	// Deprecated
+	postRestMux.HandleFunc("/rest/system/pause", s.deprecatedMakePauseHandler(true))   // device
```

Comment:
I don't think deprecating these things is moving in the right direction, to be honest. The initial GUI/API thing was to get, edit and post a complete config on every change. That's easy and it works. However it's also quite low level. From an API perspective it would make more sense to me to move in a direction where there are actions like these pause and unpause calls, similar to PUT/POST/DELETE devices and folders and toggling config options etc. So instead of removing (or deprecating) these, why not accept their existence and slowly move towards a more fine grained API? That they make persistent config changes internally is perfectly fine of course.